kola/tests/ignition/oem.go

```diff
@@ -95,7 +73,83 @@ storage:
         inline: |
-          set linux_append="flatcar.autologin"`),
+          set linux_append="flatcar.autologin"`
+)
+
+func withOldOEMMountpoint(template string) string {
+	return strings.Replace(template, oemPlaceholder, oldOEMMountpoint, -1)
+}
+
+func withNewOEMMountpoint(template string) string {
+	return strings.Replace(template, oemPlaceholder, newOEMMountpoint, -1)
+}
+
+func init() {
+	register.Register(&register.Test{
+		Name:        "cl.ignition.oem.regular",
+		Run:         reusePartitionOld,
+		ClusterSize: 1,
+		Distros:     []string{"cl"},
+		// This test overwrites the grub.cfg which does not work on cloud environments after reboot
+		Platforms:  []string{"qemu", "qemu-unpriv"},
+		MinVersion: semver.Version{Major: 3549},
+		UserData:   conf.Butane(withOldOEMMountpoint(regularButaneConfigTemplate)),
+	})
+	register.Register(&register.Test{
+		Name:        "cl.ignition.oem.regular.new",
+		Run:         reusePartitionNew,
+		ClusterSize: 1,
+		Distros:     []string{"cl"},
+		// This test overwrites the grub.cfg which does not work on cloud environments after reboot
+		Platforms:  []string{"qemu", "qemu-unpriv"},
+		MinVersion: semver.Version{Major: 3603},
+		UserData:   conf.Butane(withNewOEMMountpoint(regularButaneConfigTemplate)),
+	})
+	register.Register(&register.Test{
+		// Check new behavior from https://github.com/flatcar/bootengine/pull/58
+		// to not have to specify the OEM filesystem in Ignition
+		Name:        "cl.ignition.oem.indirect",
+		Run:         reusePartitionOld,
+		ClusterSize: 1,
+		Distros:     []string{"cl"},
+		// This test overwrites the grub.cfg which does not work on cloud environments after reboot
+		Platforms:  []string{"qemu", "qemu-unpriv"},
+		MinVersion: semver.Version{Major: 3550},
+		UserData:   conf.Butane(withOldOEMMountpoint(indirectButaneConfigTemplate)),
+	})
+	register.Register(&register.Test{
+		// Check new behavior from https://github.com/flatcar/bootengine/pull/58
+		// to not have to specify the OEM filesystem in Ignition
+		Name:        "cl.ignition.oem.indirect.new",
+		Run:         reusePartitionNew,
+		ClusterSize: 1,
+		Distros:     []string{"cl"},
+		// This test overwrites the grub.cfg which does not work on cloud environments after reboot
+		Platforms:  []string{"qemu", "qemu-unpriv"},
+		MinVersion: semver.Version{Major: 3603},
+		UserData:   conf.Butane(withNewOEMMountpoint(indirectButaneConfigTemplate)),
+	})
+	register.Register(&register.Test{
+		Name:        "cl.ignition.oem.reuse",
+		Run:         reusePartitionOld,
+		ClusterSize: 1,
+		Distros:     []string{"cl"},
+		// This test overwrites the grub.cfg which does not work on cloud environments after reboot
+		Platforms:  []string{"qemu", "qemu-unpriv"},
+		MinVersion: semver.Version{Major: 2983},
+		// Using CLC format here also covers the ign-converter case
+		UserData: conf.ContainerLinuxConfig(reuseContainerLinuxConfig),
+	})
+	register.Register(&register.Test{
+		Name:        "cl.ignition.oem.reuse.new",
```

Comment:
I wouldn't duplicate this test case because the used config is the same and only the check differs. The check doesn't have to differ because `grep /usr/share/oem/grub.cfg` should still work since `/usr/share/oem` is a symlink to `/oem`.